framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -564,9 +566,32 @@ protected void addUncheckedCodeDefaults(QualifierDefaults defs) {
                         AnnotationUtils.fromClass(elements, annotation), TypeUseLocation.OTHERWISE);
             }
         }
+    }
+
+    /**
+     * Adds standard unchecked defaults that do not conflict with previously added defaults.
+     *
+     * @param defs {@link QualifierDefaults} object to which defaults are added
+     */
+    protected void addUncheckedStandardDefaults(QualifierDefaults defs) {
         Set<? extends AnnotationMirror> tops = this.qualHierarchy.getTopAnnotations();
         Set<? extends AnnotationMirror> bottoms = this.qualHierarchy.getBottomAnnotations();
         defs.addUncheckedStandardDefaults(tops, bottoms);
+    }
+
+    /**
+     * Ensure that we have defaults specified.
+     *
+     * @param defs {@link QualifierDefaults} object to which defaults are added
+     */
+    protected void ensureDefaultsExist(QualifierDefaults defs) {
```

Comment:
The name of the method is "ensureDefaultsExist" but the method is actually checking for a default qualifier in hierarchy.

Also, is there a reason that this is in it's own method rather than in createQualifierDefaults?  Perhaps it is expected to be overridden?